### PR TITLE
Standardizes Buttons, Colours, and Fonts (part 1)

### PR DESCRIPTION
**Pull request by malsf21**
This is a pretty big PR that tries to knock out a good first half of our layout standardizing goals. It isn't a complete solution by any means, but does a lot of legwork for future (likely SASS-based) iterations. Also, merging this PR is likely a blocker to switching to SASS.

Here's what the PR does:
* When possible, removes our own custom-styled buttons with `reactstrap` buttons (and their respective colours & sizes). Not only is this good for visual consistency, but we also remove some jank code and optimize away duplicate components. 
  * Removes buttons: `EditorButton`, `SocialButton`, `SketchesButton`
* Adds Font Awesome icons to major editor buttons, and reduces the amount of text on each button. Replaces profile panel image icons (sorry Connor!).
* Standardizes colours used in navigation buttons and profile panel to the "feeling" colours in Bootstrap + our two purples. This should make our design more visually consistent.
* Swaps out Josefin & Josefin Slab (rip) for the ACM font stack of Poppins + Chivo (no Palaquin Dark needed) - more info on that [here](http://design.uclaacm.com/). Removes some extraneous fonts that we didn't use.
* "Cleans up" Profile Panel CSS by a bit, removes "Coming Soon" banners - more work needs to be done.
* Removes unneeded functions/one-line renders
* Adds link in our footer icon to our repo!

### CODE REVIEW

**Comment by connorbo97** [comment]
Rip console image, jk this one looks way better. I think some spacing between the button icon and the text would be nice.
image
versus
image

**Comment by connorbo97** [comment]
Not too big a fan of the font. In the guidelines it says

**Comment by connorbo97** [comment]
Unsure how easily this is done with reactstrap, but I feel like a different hover color on the slide out panel buttons would be nice. Also, is there a standard for aligning icons in a menu?

As far as I can tell, currently all content is centered, but I'm not sure if the icons should be left aligned with text centered or something in that vein.

**Comment by connorbo97** [comment]
Consistency wise, 'Run Code' should probably be 'Run' to match 'Save Code' being changed to 'Save'

**Comment by malsf21** [comment]
> Not too big a fan of the font. In the guidelines it says
> 
> > There aren’t really strict guidelines on how to use these typefaces—designing with them is up to your discretion. Likewise, there’s no need to adhere to these type families, as they may change in the future. That being said, when in doubt and don’t care about font specifics, use Poppins.
> 
> I feel like the site loses some character with these bolder fonts. Then again, I might just be used to it.

Yeah, I did think Josefin gave some character. I did think it was a bit thin (but that's something we can change as well), and I do like the larger visual consistency with this and the home website. I can probably revert the fonts commit and discuss that at a later date with the team, but I do kind of want to have font consistency w/ the rest of ACM.

**Comment by malsf21** [comment]
> Unsure how easily this is done with reactstrap, but I feel like a different hover color on the slide out panel buttons would be nice. Also, is there a standard for aligning icons in a menu?
> 
> As far as I can tell, currently all content is centered, but I'm not sure if the icons should be left aligned with text centered or something in that vein.

Yeah, good point - I left-aligned it with a standard spacing between the icons and I think it looks alright. Doing different text alignment values was kind of annoying (we did it last time w/ some aggressive CSS positioning), but I'm willing to do it if it looks a lot better. LMK what you think.

**Comment by malsf21** [comment]
> Unsure how easily this is done with reactstrap, but I feel like a different hover color on the slide out panel buttons would be nice. 

Yeah, I think I got a little carried away w/ the buttons. Pushed a different style, lmk if you think it's too aggressive.

File: src/components/common/ProfilePanel.js [hunk 1]
```diff
@@ -205,83 +204,47 @@ class ProfilePanel extends React.Component {
...
-      )}
-      <span className="panel-item-content">
-        <span className="panel-item-icon">
-          <img className={"panel-item-icon"} alt="house" src={House} />
-        </span>
-        <span className="panel-item-name">Editor</span>
-      </span>
-    </div>
+      <FontAwesomeIcon icon={faPencilAlt} />
+      <span> Editor</span>
```
Comment (connorbo97) [inline src/components/common/ProfilePanel.js]: need nbsp here for consistency. I would suggest changing the Button class to use a flexbox and have the span with the text flex:1 to fill the rest of the space, but I know you hate the flex bois, so you don't have to :)

Comment (malsf21) [reply src/components/common/ProfilePanel.js]: Yup, you right! And naw I don't hate the flex bois, I'll try it out.

**Pull request merged**